scanners/flutter/flutter.go

```diff
@@ -244,108 +159,129 @@ func (Scanner) ExcludedScannerNames() []string {
-		flutterInstallStep := steps.FlutterInstallStepListItem(
-			envmanModels.EnvironmentItemModel{installerUpdateFlutterKey: "false"},
-		)
-		deploySteps := steps.DefaultDeployStepList()
-
-		// primary
-		configBuilder.SetWorkflowDescriptionTo(models.PrimaryWorkflowID, primaryWorkflowDescription)
-
-		configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, prepareSteps...)
-
-		configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, flutterInstallStep)
-
-		// restore cache is after flutter-installer, to prevent removal of pub system cache
-		configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.RestoreDartCache())
-
-		if variant.test {
-			configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.FlutterTestStepListItem(
-				envmanModels.EnvironmentItemModel{projectLocationInputKey: "$" + projectLocationInputEnvKey},
-			))
+	for i, proj := range defaultProjects {
+		proj.id = i
+
+		config, err := generateConfig(models.RepoAccessUnknown, proj)
+		if err != nil {
+			return nil, err
 		}
+		configs[proj.configName()] = config
+	}
 
-		configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.SaveDartCache())
+	return configs, nil
+}
+
+func findProjectLocations(searchDir string) ([]string, error) {
+	fileList, err := pathutil.ListPathInDirSortedByComponents(searchDir, true)
+	if err != nil {
+		return nil, err
+	}
 
-		configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, deploySteps...)
+	filters := []pathutil.FilterFunc{
+		pathutil.BaseFilter("pubspec.yaml", true),
+		pathutil.ComponentFilter("node_modules", false),
+	}
 
+	paths, err := pathutil.FilterPaths(fileList, filters...)
+	if err != nil {
+		return nil, err
+	}
+
+	for i, path := range paths {
+		paths[i] = filepath.Dir(path)
+	}
+
+	return paths, nil
+}
+
+func generateConfig(repoAccess models.RepoAccess, proj project) (string, error) {
+	configBuilder := models.NewDefaultConfigBuilder()
+
+	// Common steps to all workflows
+	prepareSteps := steps.DefaultPrepareStepList(steps.PrepareListParams{RepoAccess: repoAccess})
+	flutterInstallStep := steps.FlutterInstallStepListItem(proj.flutterVersionToUse, false)
+	deploySteps := steps.DefaultDeployStepList()
+
+	// primary
+	configBuilder.SetWorkflowDescriptionTo(models.PrimaryWorkflowID, primaryWorkflowDescription)
+
+	configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, prepareSteps...)
+
+	configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, flutterInstallStep)
+
+	// restore cache is after flutter-installer, to prevent removal of pub system cache
+	configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.RestoreDartCache())
+
+	if proj.hasTest {
+		configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.FlutterTestStepListItem(
+			envmanModels.EnvironmentItemModel{projectLocationInputKey: "$" + projectLocationInputEnvKey},
+		))
+	} else {
+		configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.FlutterAnalyzeStepListItem(
```

Comment:
Isn't it useful to include the analyze step in every workflow, even if it has tests?

Reply:
It seems earlier we intentionally removed the analyze step from primary workflows: https://github.com/bitrise-io/bitrise-init/pull/223